google/gmail_chatgpt_slack/program.py

```diff
@@ -0,0 +1,293 @@
+
+    Returns: The email body as a string.
+    """
+    payload = message["payload"]
+    body = ""
+    if "parts" in payload:
+        parts = payload["parts"]
+        for part in parts:
+            if part["mimeType"] == "text/plain":
+                body = base64.urlsafe_b64decode(part["body"]["data"]).decode("utf-8")
+                break
+    else:
+        print("No parts in payload.", flush=True)
+        body = base64.urlsafe_b64decode(payload["body"]["data"]).decode("utf-8")
+    return body
+
+
+def _add_label_to_message(label_id: str, message_id: str, service):
+    """Add a label to a message in the user's inbox.
+
+    Args:
+        service: An authorized Gmail API service instance.
+        message_id: The email message_id as a string.
+        label_id: The label_id as a string.
+    """
+    try:
+        modified_message = (
+            service.users()
+            .messages()
+            .modify(userId="me", id=message_id, body={"addLabelIds": [label_id]})
+            .execute()
+        )
+
+        print("Message modified successfully:", modified_message)
+    except Exception as e:
+        print("An error occurred:", e)
+
+
+def _create_label(service, label_name: str) -> dict:
+    """Create a new label in the user's gmail account.
+
+    Args:
+        service: An authorized Gmail API service instance.
+        label_name: The name of the label to be created.
+
+    Returns: The created label as a dictionary.
+    https://developers.google.com/gmail/api/reference/rest/v1/users.labels#Label
+    """
+    label = {
+        "labelListVisibility": "labelShow",
+        "messageListVisibility": "show",
+        "name": label_name,
+    }
+
+    try:
+        created_label = (
+            service.users().labels().create(userId="me", body=label).execute()
+        )
+        print(f"Label created: {created_label['name']}")
+        return created_label
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        return None
+
+
+def _get_label_id(service, label_name: str) -> str:
+    """Get the label_id for a label with the provided name if it exists.
+    Args:
+        service: An authorized Gmail API service instance.
+        label_name: The name of the label to retrieve the id for.
+    """
+    try:
+        labels_response = service.users().labels().list(userId="me").execute()
+        labels = labels_response.get("labels", [])
+        for label in labels:
+            if label["name"] == label_name:
+                return label["id"]
+        print(f'Label "{label_name}" does not exist.')
+        return None
+    except Exception as e:
+        print("An error occurred while retrieving labels:", e)
+        return None
+
+
+def _get_message_count(service) -> int:
+    """Get the total number of messages in the user's inbox.
+
+    Args:
+        service: An authorized Gmail API service instance.
+
+    Returns: The total number of messages as an integer.
+    """
```

Comment:
Generally speaking, if a function is both short and easy to understand, it's reasonable to use a 1-line docstring instead of a full text block.

```suggestion
    """Returns The total number of messages in the user's inbox."""
```